File: Backend/Blackbox_Tests/Topictests.cs
class tests
{
   public async Task CreateTopic_ValidInput_ShouldReturnSuccess()
    {
        var client = new HttpClient(); // Or use WebApplicationFactory for in-memory testing
        var topic = new { Title = "Microservices", Description = "Intro to distributed systems" };
        var response = await client.PostAsJsonAsync("https://localhost:5001/api/topics", topic);
        Assert.True(response.IsSuccessStatusCode);
    }
}

File: Backend/Blackbox_Tests/Topictests.js
describe('Topic Creation', () => {
  it('should allow a student to create a topic', () => {
    cy.visit('http://localhost:3000');
    cy.get('#title').type('Microservices 101');
    cy.get('#description').type('Intro to distributed systems');
    cy.get('#submit-topic').click();
    cy.contains('Microservices 101').should('exist');
  });
});
